src/pages/KpiDashboard.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react'
+import { fetcher } from '../utils'
+import EChartsReact from 'echarts-for-react'
+
+export default function KpiDashboard() {
+    const [data, setData] = useState([])
+    const [condition, setCondition] = useState({ method: 'year' })
+    const [xAxis, setxAxis] = useState<string[]>([])
+    const [yAxis, setyAxis] = useState<number[]>([])
+    const [option, setOption] = useState<object>({
+        xAxis: {
+            type: 'category',
+            data: xAxis
+        },
+        yAxis: {
+            type: 'value'
+        },
+        series: [
+            {
+                data: yAxis,
+                type: 'line'
+            }
+        ]
+    })
+    const getData = async () => {
+        let OriginalData = await fetcher("epidemic/cases_malaysia.csv")
+        console.log(OriginalData);
+        const xArr: string[] = []
+        const yArr: number[] = []
+        OriginalData = OriginalData.filter((item: any) => {
+            const itemDate: string = item.date
+            const itemDateArr = itemDate.split('-')
+            const itemYear: number = Number(itemDateArr[0])
+            const itemMonth: number = Number(itemDateArr[1])
+            const itemDay: number = Number(itemDateArr[2])
+
+            const currentDate = new Date()
+            const currentYear = currentDate.getFullYear()
+            const currentMonth: number = currentDate.getMonth() + 1
+
+            switch (condition.method) {
+                case 'year':
+                    if (itemYear === currentYear) {
+                        xArr.push(itemDate)
+                        yArr.push(item.cases_new)
+                    }
+                    return itemYear === currentYear
+                case 'month':
+                    if (itemYear === currentYear && itemMonth === currentMonth) {
+                        xArr.push(itemDate)
+                        yArr.push(item.cases_new)
+                    }
+                    return itemYear === currentYear && itemMonth === currentMonth
+                default:
+                    break;
+            }
+        })
+        setxAxis(xArr)
+        setyAxis(yArr)
+    }
+    useEffect(() => {
+        getData()
+    }, [])
```

Comment:
Same here bro, I suggest putting it in a loader function so we get better performance